src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/article/Helper/Menu/MenuAdaptor.php

```diff
@@ -13,19 +13,23 @@ class {{ entity_class }}MenuAdaptor implements MenuAdaptorInterface
 {
     private $overviewpageIds = array();
 
+    private $em;
+
     /**
      * @param EntityManager $em The entity manager
      */
     public function __construct(EntityManager $em)
     {
-	$overviewPageNodes = $em->getRepository('KunstmaanNodeBundle:Node')->findByRefEntityName('{{ namespace|replace({"\\": "\\\\"}) }}\\Entity\\Pages\\{{ entity_class }}OverviewPage');
-        foreach ($overviewPageNodes as $overviewPageNode) {
-            $this->overviewpageIds[] = $overviewPageNode->getId();
-        }
+	    $this->em = $em;
     }
 
     public function adaptChildren(MenuBuilder $menu, array &$children, MenuItem $parent = null, Request $request = null)
     {
+        $overviewPageNodes = $this->em->getRepository('KunstmaanNodeBundle:Node')->findByRefEntityName('{{ namespace|replace({"\\": "\\\\"}) }}\\Entity\\Pages\\{{ entity_class }}OverviewPage');
+        foreach ($overviewPageNodes as $overviewPageNode) {
+            $this->overviewpageIds[] = $overviewPageNode->getId();
```

Comment:
The overviewpageIds should be cached in private class variable because the adaptChildren is called multiple times